### PR TITLE
mailer client not taking account configuration update in hot reload

### LINKED ISSUE #3038: Mail Client not taking account configuration update in hot reload
Everything is in the title.

### PR DESCRIPTION
**Pull request by machi1990**
Fixes #3038

/cc @cescoffier

### CODE REVIEW

**Review by cescoffier — APPROVED** [review]

**Comment by cescoffier** [comment]
Waiting for CI.

File: extensions/mailer/runtime/src/main/java/io/quarkus/mailer/impl/MailConfigRecorder.java
Comment (gsmet) [inline extensions/mailer/runtime/src/main/java/io/quarkus/mailer/impl/MailConfigRecorder.java]: @cescoffier @machi1990 I'm not totally convinced by this change.

Shouldn't we properly close the client on shutdown and start a new one each time?

**Review by gsmet — CHANGES_REQUESTED** [review]
Marking it as "Request changes" to get an answer to my question before it being merged :).

File: extensions/mailer/runtime/src/main/java/io/quarkus/mailer/impl/MailConfigRecorder.java
Comment (machi1990) [reply extensions/mailer/runtime/src/main/java/io/quarkus/mailer/impl/MailConfigRecorder.java]: > Shouldn't we properly close the client on shutdown and start a new one each time?

This was my other option but creating a client on each reload even though unnecessary was not right to me.

Comment (gsmet) [reply extensions/mailer/runtime/src/main/java/io/quarkus/mailer/impl/MailConfigRecorder.java]: Well, we usually don't do fancy things like that in other extensions and TBH I'm not sure I like it.

The contract we have for dev mode was: let's restart your Quarkus fully. And if we begin having all sorts of edge cases like that, I'm not sure it won't be a Pandora box of future bugs.

Comment (gsmet) [reply extensions/mailer/runtime/src/main/java/io/quarkus/mailer/impl/MailConfigRecorder.java]: But I'll let @cescoffier decide, I just wanted to raise my concerns.

Comment (machi1990) [reply extensions/mailer/runtime/src/main/java/io/quarkus/mailer/impl/MailConfigRecorder.java]: > The contract we have for dev mode was: let's restart your Quarkus fully. 

Sounds reasonable enough. @cescoffier WDYT?

**Comment by aloubyansky** [comment]
@cescoffier i removed waiting-for-ci set by you earlier, since your opinion is needed to clear some concerns.

**Comment by gsmet** [comment]
@cescoffier ping? I think we can converge quickly on this one, could you take a look at my comments?

**Comment by machi1990** [comment]
I have added a second commit as par Guillaume suggestion on completely re-initialising the client during restart. The same approach was applied here: https://github.com/quarkusio/quarkus/pull/3110.

Before merging we can squash the two commits into one atomic commit.

**Review by gsmet — APPROVED** [review]

**Comment by gsmet** [comment]
I'm merging this. @cescoffier can still revert the second commit if he disagrees with my approach.

**Comment by gsmet** [comment]
Thanks @machi1990 !

**Pull request merged**
